### DISCUSSION TITLE
How to change the charging mechanism when the override mechanism is being removed.

### DISCUSSION THREAD

**Discussion by frodeheg** [frodeheg/no.sparegris#97] [General]
The current plan to change the basic charging algorithm is as follows:
---
Automatically get info from the Easee app on whether or not 3-phase charging is available **(requires changes to the Easee app).**

* If 3-phase 230V:
  * Stop current = Pause current = 0 A
  * Minimum charging current = 11 A
* If 1-phase 230V or n-phase 400V:
  * Stop current = Pause current = 4 A
  * Minimum charging current = 7 A

Fill in your comments below to give your input on how to change it before the override mechanism is removed.

Unsure about the Peugeot bug that disconnect the charger below 6A. If it pauses charging at 6A then maybe 6A should be set as Pause current and 4A as Stop current?

**Comment by frodeheg** [comment]
Some background can be found in these tickets:
https://github.com/frodeheg/no.sparegris/issues/92
https://github.com/frodeheg/no.sparegris/issues/94
https://community.homey.app/t/app-pro-piggy-bank/66677/304?u=frodeheg

Will it work better if always regulating as if it is 1-phase, then if a lot of power is available and we have reached the roof of 1-phase then briefly set the current to 0 A and then jump to 11A and increase from there?

**Comment by frodeheg** [comment]
Apparently the Tesla Model Y need the minimum charging current to be 11A to start charging... but it can be as low as 5A after.
Thus, it might be better to use 11A as a default starting current and then adjust from there, this will keep the charging at 3-phases too...

**Reply by ThomasEdvardsen** [reply]
I don't think this is right. But if you use an Easee charger, it will not start 3-phase charging if the starting current is less than 11A. Can start with 7A, but then Easee only gives 1-phase. My Tesla Model Y does not have any requirements for starting current of 11A.

**Comment by frodeheg** [comment]
The following ticket is the first attempt to fix the issue (override is still kept though): #148

**Comment by frodeheg** [comment]
Added an override for max current in ticket #150 
I also removed the reference to this thread and the comment that the override would be removed. I suppose it can stay there for anyone to play with it
